common/subtitle-coloring/subtitle-coloring.ts

```diff
@@ -0,0 +1,790 @@
+import { Fetcher, RichSubtitleModel } from '@project/common';
+import { Anki } from '@project/common/anki';
+import {
+    AsbplayerSettings,
+    DictionaryTrack,
+    getFullyKnownTokenStatus,
+    TokenMatchStrategy,
+    TokenMatchStrategyPriority,
+    TokenStatus,
+    TokenStyling,
+} from '@project/common/settings';
+import { SubtitleCollection, SubtitleCollectionOptions } from '@project/common/subtitle-collection';
+import { arrayEquals, filterAsync, inBatches } from '@project/common/util';
+import { Yomitan } from '@project/common/yomitan/yomitan';
+
+const TOKEN_CACHE_BUILD_AHEAD = 10;
+const TOKEN_CACHE_BATCH_SIZE = 1; // Processing more than 1 at a time is slower
+const TOKEN_CACHE_ERROR_REFRESH_INTERVAL = 10000;
+const ANKI_RECENTLY_MODIFIED_INTERVAL = 10000;
+const HAS_LETTER_REGEX = /\p{L}/u;
+
+interface TrackState {
+    track: number;
+    dt: DictionaryTrack;
+    yomitanHealthy: boolean;
+    tokenizeCache: Map<string, string[]>;
+    tokenStatusCache: Map<string, TokenStatus | null>;
+    lemmatizeCache: Map<string, string[]>;
+    ankiCardIdStatuses: Map<number, TokenStatus>;
+    ankiSuspendedCardIds: Set<number>;
+}
+
+export class SubtitleColoring extends SubtitleCollection<RichSubtitleModel> {
```

Comment:
Just comment, don't need to do anything. Composition instead of inheritance would have been possible if you wrote another class that ties the `subtitle coloring` to the `subtitle collection`. Or just made `subtitle collection` a wrapper around `subtitle collection`, but I know you've already said that you don't like mixing responsibilities up like that.